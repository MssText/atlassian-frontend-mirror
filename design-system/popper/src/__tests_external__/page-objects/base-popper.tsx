import {
  CypressType,
  InProductTestPageObject,
} from '@atlaskit/in-product-testing';

export class BasePopperPageObject extends InProductTestPageObject {
  constructor(cy: CypressType) {
    super(cy);
  }

  public assertPopperStyle = (
    elementSelector: keyof HTMLElementTagNameMap | string,
  ) => {
    return this.cy.get(elementSelector).should('be.visbile');
  };
}
