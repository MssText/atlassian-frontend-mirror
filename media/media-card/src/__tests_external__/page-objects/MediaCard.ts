// complains about devDependencies instead of dependencies
// eslint-disable-next-line import/no-extraneous-dependencies
import { cy } from 'local-cypress';

export class MediaCardPageObject {
  constructor(private cy: Cypress.cy) {}

  public expectCardReady(numOfCards: number) {
    return this.cy
      .get('[data-test-media-name]')
      .should('have.length', numOfCards)
      .each((element) => {
        cy.wrap(element)
          // NOTE: `be.visible` is used to ensure the file is still visible
          // whilst it is being uploaded.
          .should('be.visible')
          .should('have.attr', 'data-test-status', 'complete');
      });
  }

  public findCardNth(nth: number) {
    return this.cy
      .get('[data-test-media-name][data-test-status="complete"]')
      .should('have.length.at.least', nth)
      .should('be.visible')
      .eq(nth - 1);
  }
}
