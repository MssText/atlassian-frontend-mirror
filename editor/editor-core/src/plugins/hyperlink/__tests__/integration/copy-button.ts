import { _getCopyButtonTestSuite } from '../../../copy-button/__tests__/integration/_getCopyButtonTestSuite';
import basicHyperlinkAdf from '../__fixtures__/basic-hyperlink.adf.json';
import { hyperlinkSelectors } from '../../../../__tests__/__helpers/page-objects/_hyperlink';

_getCopyButtonTestSuite({
  nodeName: 'Hyperlink',
  editorOptions: { defaultValue: basicHyperlinkAdf },
  nodeSelector: hyperlinkSelectors.hyperlink,
});
