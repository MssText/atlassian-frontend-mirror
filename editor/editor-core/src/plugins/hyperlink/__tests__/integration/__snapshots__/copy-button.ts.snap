// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy block with floating toolbar copy button 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "__confluenceMetadata": null,
                "href": "http://atlassian.com",
              },
              "type": "link",
            },
          ],
          "text": "http://atlassian.comhttp://atlassian.com",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
